src/renderer/Dataframe.js

```diff
@@ -164,106 +268,129 @@ export default class Dataframe {
         return { minx, maxx, miny, maxy };
     }
 
-    _getPointsAtPosition (p, viz) {
-        p = wToR(p.x, p.y, { center: this.center, scale: this.scale });
+    _getViewportPoints (viewportAABB) {
+        return [
+            { x: viewportAABB.minx, y: viewportAABB.miny },
+            { x: viewportAABB.minx, y: viewportAABB.maxy },
+            { x: viewportAABB.maxx, y: viewportAABB.miny },
+            { x: viewportAABB.maxx, y: viewportAABB.maxy }
+        ];
+    }
+
+    _getPointsAtPosition (pos, viz) {
+        const p = wToR(pos.x, pos.y, {
+            center: this.center,
+            scale: this.scale
+        });
+
         const points = this.decodedGeom.vertices;
         const features = [];
-        // The viewport is in the [-1,1] range (on Y axis), therefore a pixel is equal to the range size (2) divided by the viewport height in pixels
-        const widthScale = (2 / this.renderer.gl.canvas.clientHeight) / this.scale * this.renderer._zoom;
-        const vizWidth = viz.width;
-        const vizStrokeWidth = viz.strokeWidth;
 
         for (let i = 0; i < points.length; i += 2) {
             const featureIndex = i / 2;
             const center = {
                 x: points[i],
                 y: points[i + 1]
             };
-            const f = this.getFeature(featureIndex);
-            if (this._isFeatureFiltered(f, viz.filter)) {
+
+            const feature = this.getFeature(featureIndex);
+
+            if (this._isFeatureFiltered(feature, viz.filter)) {
                 continue;
             }
-            const pointWidth = vizWidth.eval(f);
-            const pointStrokeWidth = vizStrokeWidth.eval(f);
-            const diameter = Math.min(pointWidth + pointStrokeWidth, 126);
 
-            // width and strokeWidth are diameters and scale is a radius, we need to divide by 2
-            const scale = diameter / 2 * widthScale;
+            const strokeWidthScale = this._computePointWidthScale(feature, viz);
+
             if (!viz.symbol._default) {
                 const offset = viz.symbolPlacement.eval();
-                center.x += offset[0] * scale;
-                center.y += offset[1] * scale;
+                center.x += offset[0] * strokeWidthScale;
+                center.y += offset[1] * strokeWidthScale;
             }
 
-            const inside = _pointInCircle(p, center, scale);
+            const inside = pointInCircle(p, center, strokeWidthScale);
+
             if (inside) {
                 features.push(this.getFeature(featureIndex));
             }
         }
+
         return features;
     }
 
     _getLinesAtPosition (pos, viz) {
-        return this._getFeaturesFromTriangles(pos, viz.width, viz.filter);
+        return this._getFeaturesFromTriangles('line', pos, viz);
```

Comment:
Couldn't the caller just call directly to `_getFeaturesFromTriangles`, avoiding the need of having `_getLinesAtPosition` and `_getPolygonAtPosition` methods?